File: src/main/java/io/github/starwishsama/namelessbot/managers/GroupConfigManager.java
package io.github.starwishsama.namelessbot.managers;

import io.github.starwishsama.namelessbot.objects.group.GroupConfig;
import lombok.Getter;
import lombok.Setter;

import java.util.LinkedHashMap;
import java.util.Map;

public class GroupConfigManager {
    @Setter
    @Getter
    private static Map<Long, GroupConfig> configMap = new LinkedHashMap<>();

    public static GroupConfig getConfig(Long groupId){
        return configMap.getOrDefault(groupId, null);
    }

    public static void addConfig(Long groupId, GroupConfig cfg) {
        if (!configMap.containsKey(groupId)) {
            configMap.put(groupId, cfg);
        }
    }

    public static void removeConfig(Long groupId) {
        if (!configMap.containsKey(groupId)) {
            configMap.remove(groupId);
        }
    }

    public static boolean isValidGroupConfig(Long groupId){
        return getConfig(groupId) != null;
    }
}

File: src/main/kotlin/io/github/starwishsama/nbot/managers/GroupConfigManager.kt
package io.github.starwishsama.nbot.managers

import io.github.starwishsama.nbot.objects.group.GroupConfig

object GroupConfigManager {
    var configs = mutableMapOf<Long, GroupConfig>()

    fun getConfig(groupId: Long) : GroupConfig {
        return configs.getOrDefault(groupId, GroupConfig(groupId))
    }
}